packages/pages/src/common/src/parsers/sourceFileParser.test.ts

```diff
@@ -0,0 +1,134 @@
+import path from "path";
+import createTestSourceFile from "../../../util/createTestSourceFile.js";
+import SourceFileParser from "./sourceFileParser.js";
+
+describe("getDefaultExport", () => {
+  it("correctly gets default export's name when function", () => {
+    const parser = createParser(
+      `export const no = false; export default function test() {}`
+    );
+    const defaultExport = parser.getDefaultExport();
+    expect(defaultExport).toBe("test");
+  });
+  it("correctly gets default export's name when variable", () => {
+    const parser = createParser(`const test = 5; export default test`);
+    const defaultExport = parser.getDefaultExport();
+    expect(defaultExport).toBe("test");
+  });
+});
+
+describe("addDefaultExport", () => {
+  it("correctly adds default export to file", () => {
+    const parser = createParser(``);
+    parser.addDefaultExport("test");
+    expect(parser.getDefaultExport()).toBe("test");
+  });
+});
+
+describe("getExpressionByName", () => {
+  it("correctly returns variable expression when given name", () => {
+    const parser = createParser(`const foo = 5; const bar = 7;`);
+    const expression = parser.getExpressionByName("foo");
+    expect(expression).toBe("const foo = 5;");
+  });
+  it("correctly returns function expression when given name", () => {
+    const parser = createParser(`function foo(){} const bar = 7;`);
+    const expression = parser.getExpressionByName("foo");
+    expect(expression).toBe("function foo(){}");
+  });
+  it("correctly returns empty string when expression doesn't exist", () => {
+    const parser = createParser(`const test = 5;`);
+    const expression = parser.getExpressionByName("foo");
+    expect(expression).toBe("");
+  });
+});
+
+describe("addExpressions", () => {
+  it("correctly adds expression to file", () => {
+    const parser = createParser(``);
+    parser.addExpressions(["const test = 5;"]);
+    expect(parser.getExpressionByName("test")).toBe("const test = 5;");
+  });
+});
+
+describe("getAllImports", () => {
+  it("correctly gets all imports from file", () => {
+    const parser = createParser(
+      `import * as React from "react"; import Template from "@yext/pages";`
+    );
+    const imports = parser.getAllImports();
+    expect(imports[0].moduleSpecifier).toBe("react");
+    expect(imports[1].moduleSpecifier).toBe("@yext/pages");
+  });
+  it("handles file having no imports", () => {
+    const parser = createParser(`const test = 5;`);
+    const imports = parser.getAllImports();
+    expect(imports).toEqual([]);
+  });
+  it("handles multiple named imports from one path", () => {
+    const parser = createParser(
+      `import {Template, TemplateConfig, TemplateProps} from "@yext/pages";`
```

Comment:
Does this work when there's a default export too?

`import Foo, {Template, TemplateConfig, TemplateProps} from "@yext/pages";`